app/services/fees/calculator.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2021 LimeChain Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fees
+
+import (
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/limechain/hedera-eth-bridge-validator/app/domain/provider"
+	"github.com/limechain/hedera-eth-bridge-validator/app/helper"
+	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/shopspring/decimal"
+	"math"
+	"math/big"
+)
+
+type FeeCalculator struct {
+	rateProvider  provider.ExchangeRateProvider
+	configuration config.Hedera
+}
+
+func NewFeeCalculator(rateProvider provider.ExchangeRateProvider, configuration config.Hedera) *FeeCalculator {
+	return &FeeCalculator{
+		rateProvider:  rateProvider,
+		configuration: configuration,
+	}
+}
+
+func (fc FeeCalculator) ValidateExecutionFee(transferFee string, transferAmount string, gasPriceGwei string) (bool, error) {
```

Comment:
I think this should return an error only when there is an actual error.
We can introduce an additional `string` parameter with a message stating why the fee validation has failed and log it in the handler

Reply:
Isn't the purpose of the error to state `why the fee validation has failed`?